functions/venue.js

```diff
@@ -543,6 +544,41 @@ exports.toggleDustStorm = functions.https.onCall(async (_data, context) => {
   }
 });
 
+exports.getVenue = functions.https.onCall(async (data) => {
+  const { venueId } = data;
+
+  assertValidVenueId(venueId, "venueId");
+
+  // Note: we're explicitly not calling checkAuth(context) here as this function needs to work even before the user is logged in.
+
+  const venueDocumentSnapshot = await admin
+    .firestore()
+    .collection("venues")
+    .doc(venueId)
+    .get();
+
+  if (!venueDocumentSnapshot || !venueDocumentSnapshot.exists) {
+    throw new HttpsError("not-found", `Venue ${venueId} not found`);
+  }
+
+  const eventsQuerySnapshot = await admin
+    .firestore()
+    .collection("venues")
+    .doc(venueId)
+    .collection("events")
+    .orderBy("start_utc_seconds")
+    .get();
+
+  return {
+    venue: Object.assign(venueDocumentSnapshot.data(), {
+      id: venueDocumentSnapshot.id,
+    }),
+    events: eventsQuerySnapshot.docs.map((doc) =>
+      Object.assign(doc.data(), { id: doc.id })
+    ),
+  };
```

Comment:
Why was this changed from my original implementation? And why is it using `Object.assign` here? We use ES6+ spread syntax.

I don't think bundling the events here is a good idea. The whole point of this function (if even still needed) is to get the base venue config as soon as possible, not the events. And further.. the events probably shouldn't be getting exposed without auth anyways.

If you're going to keep this function, please go basic to the original basic type I had it as explicitly.

Reply:
Will deal with https://github.com/sparkletown/sparkle/pull/1713#discussion_r665027700 first, then if we decide to keep it, will do proper adjustments.

The rationale was to get any data that isn't specific to user (venue and events) and see if I could match it to the data retrieved with the original method, to keep it in parity of sort.

>  the events probably shouldn't be getting exposed without auth anyways.
Might be prudent if we get a definitive Yes/No on this, should we proceed. How sensitive is the data in that collection?

Reply:
Removed entire function in 0da80c02d45c99456fdaff937409dc563ca3fdb4